vars/dockerBuildArgs.groovy

```diff
@@ -48,27 +55,75 @@ String call(Map config = [:]) {
+          repo_name = env.DAOS_STACK_EL_7_DOCKER_RELEASE_REPO
+        } else if ((repo_type == 'dev') &&
+                   (env.DAOS_STACK_EL_7_DOCKER_DEV_REPO)) {
+          repo_name = env.DAOS_STACK_EL_7_DOCKER_DEV_REPO
+        }
+        if (repo_name) {
+          ret_str += ' --build-arg REPO_EL7=' + repo_name
+        }
       }
-      if (env.DAOS_STACK_LEAP_15_LOCAL_REPO) {
-        ret_str += ' --build-arg REPO_LOCAL_LEAP15=' +
-                   env.DAOS_STACK_LEAP_15_LOCAL_REPO
+      if (stage_info['target'] == 'centos8') {
+        if ((repo_type == 'local') && (env.DAOS_STACK_EL_8_LOCAL_REPO)) {
+            repo_name = env.DAOS_STACK_EL_8_LOCAL_REPO
+        } else if ((repo_type == 'stable') &&
+                   (env.DAOS_STACK_EL_8_DOCKER_STABLE_REPO)) {
+          repo_name = env.DAOS_STACK_EL_8_DOCKER_STABLE_REPO
+        } else if ((repo_tpe == 'release') &&
+                   (env.DAOS_STACK_EL_8_DOCKER_RELEASE_REPO)) {
+          repo_name = env.DAOS_STACK_EL_8_DOCKER_RELEASE_REPO
+        } else if ((repo_type == 'dev') &&
+                   (env.DAOS_STACK_EL_8_DOCKER_DEV_REPO)) {
+          repo_name = env.DAOS_STACK_EL_8_DOCKER_DEV_REPO
+        }
+        if (repo_name) {
+          ret_str += ' --build-arg REPO_EL8=' + repo_name
+        }
       }
-      if (env.DAOS_STACK_UBUNTU_20_04_LOCAL_REPO) {
-        ret_str += ' --build-arg REPO_UBUNTU_20_04=' + env.DAOS_STACK_UBUNTU_20_04_LOCAL_REPO
+      if (stage_info['target'] == 'leap15') {
+        if (repo_type == 'local') {
+          if (env.DAOS_STACK_LEAP_15_LOCAL_REPO) {
+            ret_str += ' --build-arg REPO_LOCAL_LEAP15=' +
+                       env.DAOS_STACK_LEAP_15_LOCAL_REPO
+          }
+          if (env.DAOS_STACK_LEAP_15_GROUP_REPO) {
+            ret_str += ' --build-arg REPO_GROUP_LEAP15=' +
+                       env.DAOS_STACK_LEAP_15_GROUP_REPO
+          }
+        } else if ((repo_type == 'stable') &&
+                   (env.DAOS_STACK_LEAP_15_DOCKER_STABLE_REPO)) {
+          repo_name = env.DAOS_STACK_LEAP_15_DOCKER_STABLE_REPO
+        } else if ((repo_tpe == 'release') &&
+                   (env.DAOS_STACK_LEAP_15_DOCKER_RELEASE_REPO)) {
+          repo_name = env.DAOS_STACK_LEAP_15_DOCKER_RELEASE_REPO
+        } else if ((repo_type == 'dev') &&
+                   (env.DAOS_STACK_LEAP_15_DOCKER_DEV_REPO)) {
+          repo_name = env.DAOS_STACK_LEAP_15_DOCKER_DEV_REPO
+        }
+        if (repo_name) {
+          ret_str += ' --build-arg REPO_LEAP=' + repo_name
+        }
       }
-      if (env.DAOS_STACK_LEAP_15_GROUP_REPO) {
-        ret_str += ' --build-arg REPO_GROUP_LEAP15=' +
-                   env.DAOS_STACK_LEAP_15_GROUP_REPO
+      if (stage_info['target'] == 'ubuntu20.04') {
+        if ((repo_type == 'local') &&
+            (env.DAOS_STACK_UBUNTU_20_04_LOCAL_REPO)) {
+              ret_str += ' --build-arg REPO_UBUNTU_20_04=' +
+                         env.DAOS_STACK_UBUNTU_20_04_LOCAL_REPO
+        }
```

Comment:
It seems like this could be a lot smaller and maintainable by first setting a distro alias such as:

```groovy
String dist = ''
if (stage_info['target'] == 'centos7') {
    dist='EL_7'
} else if (stage_info['target'] == 'centos8') {
    dist = 'EL_8'
}
...
```
and then use `${dist}` in a single block of `repo_type` `if`s.  Not sure if variables will be resolved in the env. var name, but worth trying perhaps to get this huge block under a bit more control.

There are some alternative mechanisms to [get environment variables](https://stackoverflow.com/questions/40215394/how-to-get-environment-variable-in-jenkins-groovy-script-console) that might be a bit programmatic.